signals.py

```diff
@@ -72,6 +73,10 @@ def topcoin(self, pairs, rank):
         market = self.cgvalues(rank)
 
         self.logging.debug(self.cgvalues.cache_info())
+        if isinstance(pairs, str): 
```

Comment:
@M1cha3l-2022  is instance(pairs, str) is already in the else statement. Logging should be there.